src/crypto/crypto_util.cc

```diff
@@ -65,6 +65,16 @@ MUST_USE_RESULT CSPRNGResult CSPRNG(void* buffer, size_t length) {
     if (1 == RAND_status())
       if (1 == RAND_bytes(static_cast<unsigned char*>(buffer), length))
         return {true};
+#if OPENSSL_VERSION_MAJOR >= 3
+    const auto code = ERR_peek_last_error();
+    // A misconfigured OpenSSL 3 installation may report 1 from RAND_poll()
+    // and RAND_status() but fail in RAND_bytes() if it cannot look up
+    // a matching algorithm for the CSPRNG.
+    if (ERR_GET_LIB(code) == ERR_LIB_RAND &&
+        ERR_GET_REASON(code) == RAND_R_UNABLE_TO_FETCH_DRBG) {
```

Comment:
You should probably also check for `RAND_R_UNABLE_TO_CREATE_DRBG` and `RAND_R_ERROR_INSTANTIATING_DRBG`.

Or maybe just checking `ERR_LIB_RAND` is enough; any error is likely sufficiently permanent that retrying is futile.

Reply:
Would an error for missing entropy also result in an `ERR_LIB_RAND` error? (There is some prior conversation about what exactly to test for in #46200).

Reply:
I've added the suggested extra reasons. There's a complete list in https://github.com/nodejs/node/blob/671ffd7825272bfb38ea2bd53a349f79ce01c9f8/deps/openssl/openssl/crypto/err/openssl.txt#L1088-L1132 some of which refers to entropy. 
 
> Or maybe just checking `ERR_LIB_RAND` is enough; any error is likely sufficiently permanent that retrying is futile.

I think I'd prefer to err on the side of caution in this PR -- we have reproducible test cases being fixed. Generalizing this further could be left to a follow up PR?